### DISCUSSION TITLE
知覚/探索スキルのテコ入れ

### DISCUSSION THREAD

**Discussion by habu1010** [hengband/hengband#679] [Ideas]
## 概要

現状ほぼ完全に死にスキルとなっている、知覚/探索スキルのテコ入れをしたい。

### 知覚スキル

- 罠/隠し扉の発見率
  探索コマンドや、自動探索が発生した時に、隣接マスそれぞれに対して 知覚スキル% の確率で罠/隠し扉を発見する。

### 探索スキル

- 自動探索の発動率
  移動/その場に留まるを行った時、スキルが50以上ならかならず、50以下なら 1 / (50 - スキル) の確率で探索コマンドが実行される。

- 階の雰囲気
  探索スキルが高いほど早く階の雰囲気が分かる。

**Comment by habu1010** [comment]
とりあえず現状の仕様を強化する案として一つ。
スキルが上がれば隣接マスのみでなく発見できる半径が増えていく。

**Comment by oguchi2918** [comment]
スキルに応じた効果の案として

・ESPや感知の半径にボーナス/ペナルティ(+2~-2程度)
・英雄的以上なら確率(最大50%程度)で装備無しでも警告発動(虫の予感が背中に走った！)

なんてどうでしょうか。

ステータス画面の話ですが、
知覚・探索スキルは片方だけ上下させることが不可能なので、
知覚スキルの方は現在の表示形式を消して別の場所に罠・隠し扉発見率％と明記した方がよいかと思います。

**Comment by deskull-m** [comment]
個人的にバリアントで考えた案なのですが
* アイテム感知とは別に難度の高い判定を用意してそれが通ると別枠でアイテム見つかる
* 敵を殺したときに「目星がきくから」ドロップするアイテムが増えるとする。
位強化しようかと考えています。

**Comment by habu1010** [comment]
まず前提として、知覚スキルと探索スキルが現状隠された物を発見するという、同じ方面のベクトルのスキルとしてダブっているので、これを分離したほうがいいような気がします。幸い、これらのスキルが関わっている現象は最初に記載したとおり非常に少ないので、新しい仕様の導入は比較的しやすそうです。

方針としては探索スキルは現状の罠や隠し扉を発見する能力とし、知覚スキルを危険やモンスター/アイテムの存在等を察知する能力に分離するのがいいのではないかと思います。

その方針で行くと、ここまで上がっている案では、

- 知覚スキル
  - ESPや感知の範囲の増減
  - 警告能力の発動
  - 階の雰囲気が分かるまでの速さ

- 探索スキル
  - スキルが高いと離れた場所の罠/隠し扉も発見、発見率にも影響
  - 敵を倒した時にドロップアイテムが増えることがある
  - アイテム感知で別枠でアイテムが見つかる（これは仮に実装するとしたら慎重に仕様を考えないと無闇なスカムの対象となってしまいそう）
 
のようにできそうです。

**Reply by habu1010** [reply]
知覚スキル/探索スキルを分離する場合

- 現在探索能力+のアイテムで知覚スキル/探索スキルの両方に影響がある仕様をどうするか
  1. 現状維持
  2. 探索スキルのみに影響
  3. 探索スキルのみに影響し、知覚スキルに影響するフラグを新設
